### PR TITLE
[DS18x20] Enhance use of aliases

### LINKED ISSUE #6970: Rename Sensors / give them friendly names
Hello, I have three DS18B20 connected and working perfectly.

However, the names are DS18B20 -1, DS18B20-2, DS18B20-3

How can these be renamed to friendly names? I am assuming they cant be because I was not able to find any info about this what-so-ever.

Might the only way this is possible is compiling my own build with the names embedded?

I think the feature request in that case is obvious.
Thank you
Ryan

**Have you looked for this feature in other issues and in the docs?**  

**Is your feature request related to a problem? Please describe.**  
_A clear and concise description of what the problem is._

**Describe the solution you'd like**  
_A clear and concise description of what you want to happen._

**Describe alternatives you've considered**  
_A clear and concise description of any alternative solutions or features you've considered._

**Additional context**  
_Add any other context or screenshots about the feature request here._

**(Please, remember to close the issue when the problem has been addressed)**

### PR DESCRIPTION
**Pull request by SteWers**
## Description:

`#define DS18x20_USE_ID_AS_NAME`: Always show part of the address, even for one sensor 
`#define DS18x20_USE_ID_ALIAS`: The command `DS18Alias` can now be use with alphanumeric aliases, which replace the sensor name

**Related issue (if applicable):** fixes #6970 

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] The code change is tested and works with Tasmota core ESP8266 V.2.7.4.9
  - [x] The code change is tested and works with Tasmota core ESP32 V.2.0.10
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

### CODE REVIEW

File: tasmota/tasmota_xsns_sensor/xsns_05_ds18x20.ino [hunk 1]
```diff
@@ -62,7 +63,7 @@ struct {
   uint8_t valid;
   int8_t pins_id;
 #ifdef DS18x20_USE_ID_ALIAS
-  uint8_t alias;
+  char alias[17] = "0";
```
Comment (arendst) [inline tasmota/tasmota_xsns_sensor/xsns_05_ds18x20.ino:66]: That's a costly business. Time you make this struct virtual to save lots of memory. Rewrite to use calloc()

File: tasmota/tasmota_xsns_sensor/xsns_05_esp32_ds18x20.ino [hunk 2]
```diff
@@ -61,7 +62,7 @@ struct {
   uint8_t valid;
   int8_t pins_id;
 #ifdef DS18x20_USE_ID_ALIAS
-  uint8_t alias;
+  char alias[17] = "0";
```
Comment (arendst) [inline tasmota/tasmota_xsns_sensor/xsns_05_esp32_ds18x20.ino:65]: Ditto. See above.

File: tasmota/tasmota_xsns_sensor/xsns_05_ds18x20.ino (hunk 1 above)
Comment (SteWers) [reply tasmota/tasmota_xsns_sensor/xsns_05_ds18x20.ino]: Done

File: tasmota/tasmota_xsns_sensor/xsns_05_esp32_ds18x20.ino (hunk 2 above)
Comment (SteWers) [reply tasmota/tasmota_xsns_sensor/xsns_05_esp32_ds18x20.ino]: Done

**Comment by SteWers** [comment]
Is still something to change?

**Pull request merged**